core/variables.js

```diff
@@ -472,15 +472,11 @@ exports.getOrCreateVariablePackage = getOrCreateVariablePackage;
  * @alias Blockly.Variables.getVariable
  */
 const getVariable = function(workspace, id, opt_name, opt_type) {
-  const potentialVariableMap = workspace.getPotentialVariableMap();
   let variable = null;
   // Try to just get the variable, by ID if possible.
   if (id) {
-    // Look in the real variable map before checking the potential variable map.
+    // Look in the variable map
```

Comment:
The main idea behind this refactor is that for any workspace, we have two properties: variableMap and potentialVariableMap. "Real" workspaces (not flyouts) use the variableMap but not the potentialVariableMap and flyout workspaces use the potentialVariableMap but not the variableMap (mostly, see below). We could simplify this logic by just using the variableMap for both use cases. We still need the concept of "potential" variables that exist in the flyout but not the main workspace, but we can store those in the same property to simplify things.

The flyout was using the variableMap property to store the variableMap for the target workspace. However, you can just access that via `workspace.targetWorkspace.getVariableMap()` which frees you up to use the `variableMap` property for the "potential" variables.

For this part, you still need to do what the comment you deleted says: look in the real variable map before checking for potential variables. In our new code, that means if we're in a flyout, we need to check the target workspace's variable map first (real variables) and then check the flyout's variable map (potential variables) if needed. If we're not in a flyout, just check the variable map (real variables).

So you can do something like

```js
if (workspace.isFlyout) {
  // Check for real variables in the target workspace first.
  variable = workspace.targetWorkspace.getVariableById(id);
}
if (!variable) {
  // Either this is a flyout and the main workspace didn't contain the variable, so check for potential variables, or this wasn't a flyout workspace so just check for real ones.
  variable = workspace.getVariableById(id);
}
```

So looking at this logic, I'm not sure that we should follow through with this refactor. It simplifies things in one respect, because we are eliminating the `potentialVariableMap` property which is unused for main workspaces. However, now we are using one property to mean two different things depending on whether it's a flyout workspace or not, which is a different kind of complexity that comes through on the comment for that second `if` statement.